src/proxy/http/metrics/service.rs

```diff
@@ -413,16 +426,11 @@ where
     }
 
     fn poll_data(&mut self) -> Poll<Option<Self::Data>, h2::Error> {
-        let poll = self.inner.poll_data().map_err(|e| self.measure_err(e));
-        let frame = try_ready!(poll);
+        let frame = try_ready!(self.inner.poll_data().map_err(|e| self.measure_err(e)));
 
-        if self.first_byte_at.is_none() {
-            self.first_byte_at = Some(clock::now());
-        }
-
-        if let c @ Some(_) = self.class_at_first_byte.take() {
-            self.record_class(c);
-            self.classify = None;
+        if !self.latency_recorded {
+            self.record_latency();
+            self.latency_recorded = true;
```

Comment:
Nit/TIOLI: Since `record_latency` takes `&mut self`, it seems reasonable to assume that _it_ would be responsible for setting this flag...